solutions/cf-702C.mdx

```diff
@@ -327,4 +327,52 @@ int main()
 
 </CPPSection>
 
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class Main {
+	//CodeSnip{Kattio}
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+
+		int n = io.nextInt(), m = io.nextInt();
+		int[] cities = new int[n];
+		TreeSet<Integer> towers = new TreeSet<>();
+		for (int i = 0; i < n; i++)	
+		{
+			cities[i] = io.nextInt();
+		}
+
+		for (int i = 0; i < m; i++)		
+		{
+			int tower = io.nextInt();
+			towers.add(tower);
+		}
+
+		int r = 0;
+		for (int i = 0; i < n; i++)		
+		{
+			int distance = (int) 2e9+1;
+			Integer closestTower = towers.lower(cities[i]), farthestTower = towers.higher(cities[i]);
```

Comment:
1. explain why you use Integer and not int
2. put declarations on separate lines

Reply:
The reason i'm using Integer instead of int is to make sure that it has a valid value and null does not appear on int and the TreeSet class generally takes in values in the form of Integer and an "Integer" cannot be converted to "int".

Reply:
I've added in all the other changes that you've told me to. Please let me know if I've missed anything and if something seems wrong. Thanks!

Reply:
yeah uh can you put your reason for using `Integer` in the
in a comment
plz & thx

Reply:
Let me know if the comment can be written in a better way or if you find any errors.